Add a "open in disposable VM" to desktop environment

ddevz opened the pull request:
This is adding a "open in disposable VM" option to desktop environment.  For example, when using a file browser, right clicking then selecting "open with" would give opening in a disposable VM as a option.  file browsers are not the intended target as they already have this functionality built in, but this makes it a option for 

this allows things like: 
`xdg-settings set default-web-browser qvm-open-in-dvm.desktop`
to be done by the user.
This does not change any any of the system defaults and it does not get used automatically

deeplow commented on qubes-rpc/qvm-open-in-dvm.desktop:
```diff
@@ -0,0 +1,10 @@
+Name=qvm-open-in-dvm
+Exec=qvm-open-in-dvm %u
+Type=Application
+Categories=Network;WebBrowser;
```
is it possible not to specify the "browser category"? I can image in the future this default `.desktop` file being more broadly used.

deeplow replied:
Perhaps it doesn't make much difference since AppVMs don't really have a menu which can sort by category. But I thought I'd mention it anyways.

ddevz replied:
well this is to solve this issue:
#https://github.com/QubesOS/qubes-issues/issues/6586
with opening URLS.

Can you specify as to which ".desktop" file you are referring to as default (this .desktop file is *not* set as default for anything.  You must manually set it).  Also can you elaborate the "more broadly used" part as in more broadly used being a good thing or a bad thing?

ddevz replied:
oh, is "Categories" a qubes thing?  I assumed it was a desktop/gnome/kde thing

deeplow replied:
Not sure

ddevz replied:
I got the .desktop file from https://github.com/SvenSemmler .  He had posted the file as part of a resolution for issue https://github.com/QubesOS/qubes-issues/issues/6586 .  (I realized it was of general use for many people, not just that issue, and am submitting it for inclusion)

Perhaps he knows?

deeplow replied:
@SvenSemmler

SvenSemmler replied:
I am super confused, why is this in 'qubes-rpc'?

ddevz replied:
qubes-rpc?  I'm not familiar with that, but I'm new so my knowledge of the codebase is limited.  I put it in qubes-core-agent-linux cause it's essentially just a wrapper for "qvm-open-in-dvm", and that's where qvm-open-in-dvm is located.  Is there a different repo that is automatically installed inside the VMs that I should have put it in?

SvenSemmler replied:
@ddevz @deeplow .. I am sorry but I don't think I can be of help in this context. My recommendation would be to send an email to qubes-devel describing what your intention is and asking for guidance where to locate the PR. @andrewdavidwong please advise if there is a better course of action.

ddevz replied:
I found a repository:  
https://github.com/QubesOS/qubes-gui-agent-linux

It has no README.md but inferring from the name, maybe this is the correct place to put it?  If so, I can move it there.

ddevz replied:
I'm sorry.  I just realized that you were asking why i was putting it in that directory.  I thought you were asking why I was putting it in that repository (out of the 151 to choose from).

You are correct.  I should have put it in /applications-dropins/
I'll change it

marmarek commented on applications-dropins/qvm-open-in-dvm.desktop:
```diff
@@ -0,0 +1,10 @@
+### Note: With this installed, typing "xdg-settings set default-web-browser qvm-open-in-dvm.desktop" will make it so that in gnome-terminal
```
This directory is for overrides of third-party menu entries. Put this one in the `app-menu` dir.

marmarek commented on qubes-rpc/Makefile:
```diff
@@ -88,3 +89,5 @@ install:
 	install -t $(DESTDIR)$(QUBESCONFDIR)/post-install.d -m 0644 post-install.d/README
 	install -t $(DESTDIR)$(QUBESCONFDIR)/post-install.d post-install.d/*.sh
+	install -d $(DESTDIR)/$(APPLICATIONSDIR)
+	install -t $(DESTDIR)/$(APPLICATIONSDIR) -m 0644 qvm-open-in-dvm.desktop
```
And also move this makefile fragment together with the file.

ddevz commented:
Ok, moved to app-menu and refactored the commits to be readable

ddevz commented:
> Ok, moved to app-menu and refactored the commits to be readable

After I commited, I noticed the updates to QubesOS/qubes-core-agent-linux - master, so used the "fetch upstream" button after the commit, but it looks like it it did a merge instead of a rebase.  Let me know if this screws anything up.

marmarek commented:
It seems to _almost_ correctly did the rebase. But "committer" name is wrong: ` ddevz authored and Your Name committed`

marmarek commented:
And the commits signature is gone.

marmarek commented:
Plus, you need to include the file in packaging files - spec for rpm, and `debian/qubes-core-agent.install` for deb.

marmarek commented:
PipelineRetry
